osv/osv.go

```diff
@@ -0,0 +1,159 @@
+package osv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/aquasecurity/vuln-list-update/utils"
+	"github.com/cheggaaa/pb"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+)
+
+const (
+	securityTrackerURL = "https://osv-vulnerabilities.storage.googleapis.com/%s/all.zip"
+	osvDir             = "osv"
+)
+
+var defaultEcosystemDirs = map[string]string{
+	"PyPI":      "python",
+	"Go":        "go",
+	"crates.io": "rust",
+}
+
+type options struct {
+	url           string
+	dir           string
+	ecosystemDirs map[string]string
+}
+
+type option func(*options)
+
+type Osv struct {
+	opts  *options
+	AppFs afero.Fs
+}
+
+func WithURL(url string) option {
+	return func(opts *options) {
+		opts.url = url
+	}
+}
+
+func WithDir(dir string) option {
+	return func(opts *options) {
+		opts.dir = dir
+	}
+}
+
+func WithEcosystem(ecosystemDir map[string]string) option {
+	return func(opts *options) {
+		opts.ecosystemDirs = ecosystemDir
+	}
+}
+
+func NewOsv(opts ...option) Osv {
+	ecosystemDirs := make(map[string]string)
```

Comment:
this should be used only if `WithEcosystem()` hasn't been called.

Reply:
added check

Reply:
Just call all `options` and then check if `ecosystemDirs` is still empty. If it is then assign defaults

Reply:
changed